azure/scope/machine.go

```diff
@@ -237,9 +237,17 @@ func (m *MachineScope) NICSpecs() []azure.ResourceSpecGetter {
 	isMultiNIC := len(m.AzureMachine.Spec.NetworkInterfaces) > 1
 
 	for i := 0; i < len(m.AzureMachine.Spec.NetworkInterfaces); i++ {
+		nic := m.AzureMachine.Spec.NetworkInterfaces[i]
+		nicName := ""
 		isPrimary := i == 0
-		nicName := azure.GenerateNICName(m.Name(), isMultiNIC, i)
-		nicSpecs = append(nicSpecs, m.BuildNICSpec(nicName, m.AzureMachine.Spec.NetworkInterfaces[i], isPrimary))
+
+		if nic.Name != nil && *nic.Name != "" {
```

Comment:
Same as the earlier comment, I think we could probably make `nic.Name` a `string` instead of `*string` in this case.